# CAR-SALES dataset

Dữ liệu `car-sales.csv`: bộ dữ liệu chứa thông tin các ôtô cũ ở một cửa hàng bán ô tô cũ. 

Bạn hãy thực hiện các yêu cầu sau:


### 1. Đọc dữ liệu và cho biết dữ liệu có bao nhiêu dòng, bao nhiêu cột

Đáp án: (10 dòng, 5 cột)

In [8]:
# Your code here
import pandas as pd
df = pd.read_csv("data\car-sales.csv")
pd.read_csv("data\car-sales.csv")
# df.shape

Manufacturer Colour  Odometer  Doors    Price
0       Toyota  White    150043      4   4000.0
1        Honda    Red     87899      4   5000.0
2       Toyota   Blue     32549      3   7000.0
3          BMW  Black     11179      5  22000.0
4       Nissan  White    213095      4   3500.0
5       Toyota  Green     99213      4   4500.0
6        Honda   Blue     45698      4   7500.0
7        Honda   Blue     54738      4   7000.0
8       Toyota  White     60000      4   6250.0
9       Nissan  White     31600      4   9700.0

### 2. Cho biết thuộc tính nào là thuộc tính phân loại, thuộc tính nào là thuộc tính định lượng

In [12]:
# Your code here
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10 entries, 0 to 9
Data columns (total 5 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Manufacturer  10 non-null     object 
 1   Colour        10 non-null     object 
 2   Odometer      10 non-null     int64  
 3   Doors         10 non-null     int64  
 4   Price         10 non-null     float64
dtypes: float64(1), int64(2), object(2)
memory usage: 528.0+ bytes


### 3. Trong dữ liệu có thông tin của bao nhiêu xe?

Đáp án: 10 xe

In [25]:
# Your code here
df.count()

Manufacturer    10
Colour          10
Odometer        10
Doors           10
Price           10
dtype: int64

### 4. Đếm số xe theo theo hãng

Đáp án: Toyota: 10 xe, Honda: 3 xe, Nissan: 2 xe, BMW: 1 xe

In [37]:
# Your code here
df.groupby(by = 'Manufacturer').count()


Colour  Odometer  Doors  Price
Manufacturer                                
BMW                1         1      1      1
Honda              3         3      3      3
Nissan             2         2      2      2
Toyota             4         4      4      4

### 5. Đếm số xe theo màu


In [36]:
# Your code here
df.groupby(by = 'Colour').count()

Manufacturer  Odometer  Doors  Price
Colour                                      
Black              1         1      1      1
Blue               3         3      3      3
Green              1         1      1      1
Red                1         1      1      1
White              4         4      4      4

### 6. Tổng giá trị các xe trong cửa hàng

Đáp án: 76450 USD


In [40]:
# Your code here
df.sum()

Manufacturer    ToyotaHondaToyotaBMWNissanToyotaHondaHondaToyo...
Colour              WhiteRedBlueBlackWhiteGreenBlueBlueWhiteWhite
Odometer                                                   786014
Doors                                                          40
Price                                                       76450
dtype: object

### 7. Top 3 xe có giá cao nhất


In [43]:
# Your code here
df.sort_values('Price', ascending=False).head(3)

Manufacturer Colour  Odometer  Doors    Price
3          BMW  Black     11179      5  22000.0
9       Nissan  White     31600      4   9700.0
6        Honda   Blue     45698      4   7500.0

### 8. Top 3 xe đã chạy nhiều cây số (km)nhất (Odometer)

Đáp án:
- Xe Nissan, White, 213095 km
- Xe Toyota, White, 150043 km
- Xe Toyota, Green, 99213 km

In [44]:
# Your code here
df.sort_values('Odometer', ascending=False).head(3)

Manufacturer Colour  Odometer  Doors   Price
4       Nissan  White    213095      4  3500.0
0       Toyota  White    150043      4  4000.0
5       Toyota  Green     99213      4  4500.0

### 9. Khảo sát xe của hãng Honda

- Tính tỷ lệ xe Honda trong dữ liệu của cửa hàng (0.3)
- Cho biết giá trung bình của xe hãng Honda của cửa hàng (6500)
- Cho biết các xe của hãng Honda trong cửa hàng có những màu sắc nào (Red, Blue)

In [56]:
# Your code here
ti_le = df[df['Manufacturer']=='Honda'].shape[0]/df.shape[0]
print(f"Tỷ lệ xe Honda trong dữ liệu của cửa hàng là {ti_le:.1f}")

gia_tri_trung_binh = df[df['Manufacturer']=='Honda']['Price'].mean()
print(f"Giá trung bình của xe hãng Honda của cửa hàng là {gia_tri_trung_binh:.0f}")

mau_sac = df[df['Manufacturer']=='Honda']['Colour'].unique()
print(f"Các xe của hãng Honda trong cửa hàng có những màu sắc {mau_sac}")


Tỷ lệ xe Honda trong dữ liệu của cửa hàng là 0.3
Giá trung bình của xe hãng Honda của cửa hàng là 6500
Các xe của hãng Honda trong cửa hàng có những màu sắc ['Red' 'Blue']


# 10. Tính toán các giá trị sau của  thuộc tính giá ('Price')

- Giá trung bình các xe (7645)
- Giá trị trung vị các xe (6625)
- Miền giá trị của xe (18500)
- Miền phân vị của xe (2750)
- Điểm phân vị 0.4 và 0.6 của giá ([5750, 7000])

In [64]:
# Your code here
trung_binh = df['Price'].mean()
print(f"Giá trung bình các xe là {trung_binh:.0f}")
trung_vi = df['Price'].median()
print(f"Giá trung vị các xe là {trung_vi:.0f}")
mien_gia_tri = df['Price'].max()-df['Price'].min()
print(f"Miền giá trị của xe là {mien_gia_tri:.0f}")
q1 = df['Price'].quantile(0.25)
q3 = df['Price'].quantile(0.75)
q = q3-q1
print(f"Miền phân vị của xe là {q:.0f}")


Giá trung bình các xe là 7645
Giá trung vị các xe là 6625
Miền giá trị của xe là 18500
Miền phân vị của xe là 2750


---

# TIPS dataset

Dữ liệu chứa trong file `tips.csv` chứa thông tin tiền tip của khách hàng tại một nhà hàng. Dữ liệu có các thông tin sau:

- `total_bill`: tổng hoá đơn (USD)
- `tip`: tiền tip/ tiền boa (USD)
- `sex`: giới tính người thanh toán hóa đơn
- `smoker`: có hút thuốc hay không
- `day`: ngày trong tuần
- `time`: thời gian dùng bữa(buổi sáng, buổi trưa, buổi tối)
- `size`: số lượng người tham gia

Bạn hãy thực hiện thống kê mô tả để tìm hiểu về hành vi tip của khách hàng tại nhà hàng

1. Đọc dữ liệu


In [ ]:
# Your code here

2. Xoá cột `Unnamed: 0`


In [ ]:
# Your code here

3. Cho biết dữ liệu có bao nhiêu dòng, bao nhiêu thuộc tính.


In [ ]:
# Your code here

4. Trong các thuộc tính, thuộc tính nào là thuộc tính phân loại, thuộc tính nào là thuộc tính định lượng


In [ ]:
# Your code here

5. Có bao nhiêu nam, bao nhiêu nữ. Tỷ lệ bao nhiêu?


In [ ]:
# Your code here

6. Có bao nhiêu người hút thuốc. Tỷ lệ bao nhiêu?


In [ ]:
# Your code here

7. Ngày nào trong tuần khách hàng hay đến dùng bữa nhất. Tỷ lệ bao nhiêu?


In [ ]:
# Your code here

8. Nhà hàng phục vụ bữa ăn nào nhiều nhất (Breakfast, Lunch, Dinner)?


In [ ]:
# Your code here

9. Nhóm khách hàng dùng bữa tại nhà hàng nhiều nhất là bao nhiêu người?


In [ ]:
# Your code here

10. Trung bình trong một bữa ăn nhà hàng phục vụ mấy khách?

In [ ]:
# Your code here

---